`no-undefined-types` doesn't know `Iterable` and `IteratorObject`

## Expected behavior

The [`no-undefined-types`](https://github.com/gajus/eslint-plugin-jsdoc/blob/v63.0.6/docs/rules/no-undefined-types.md) rule does not report an error if the variables `Iterable` and `IteratorObject` are used in the JSDoc.

`tsc` knows these two types and doesn't report any errors when they're used in a TypeScript file.

## Actual behavior

```
/home/regseb/testcase/index_jsdoc.js
  2:1  error  The type 'Iterable' is undefined        jsdoc/no-undefined-types
  3:1  error  The type 'IteratorObject' is undefined  jsdoc/no-undefined-types
```

## Files

### `eslint.config.js`

```javascript
import jsdoc from "eslint-plugin-jsdoc";

export default [
  {
    plugins: { jsdoc },
    rules: {
      "jsdoc/no-undefined-types": "error",
    }
  }
];
```

### `index_jsdoc.js`

```javascript
/**
 * @param {...Iterable<any>} its
 * @returns {IteratorObject<any>}
 * @see https://developer.mozilla.org/Web/JavaScript/Reference/Global_Objects/Iterator/concat
 */
const concatIteratorJSDoc = (...its) => {
    return Iterator.from(its.flat());
};
```

### `index_typescript.ts`

```typescript
const concatIteratorTypeScript = (...its: Iterable<any>[]): IteratorObject<any> => {
    return Iterator.from(its.flat());
};
```

### `package.json`

```json
{
    "name": "testcase",
    "version": "1.0.0",
    "type": "module",
    "dependencies": {
        "eslint": "10.5.0",
        "eslint-plugin-jsdoc": "63.0.6",
        "typescript": "6.0.3"
    }
}
```

## Steps

1. `npm i`
2. `npx eslint`
   ```
   /home/regseb/testcase/index_jsdoc.js
     2:1  error  The type 'Iterable' is undefined        jsdoc/no-undefined-types
     3:1  error  The type 'IteratorObject' is undefined  jsdoc/no-undefined-types

   ✖ 2 problems (2 errors, 0 warnings)
   ```
3. `npx tsc --noEmit index_typescript.ts`
   ```
   # no error
   ```

## Environment

- Node version: v26.3.1
- ESLint version: v10.5.0
- `eslint-plugin-jsdoc` version: 63.0.6

--- a/docs/rules/no-undefined-types.md
+++ b/docs/rules/no-undefined-types.md
@@ -1269,5 +1269,14 @@ const b = function (foo: number) {};
 
 /** {@link foo} */
 const b = (foo: number) => {};
+
+/**
+ * @param {...Iterable<any>} its
+ * @returns {IteratorObject<any>}
+ * @see https://developer.mozilla.org/Web/JavaScript/Reference/Global_Objects/Iterator/concat
+ */
+const concatIteratorJSDoc = (...its) => {
+    return Iterator.from(its.flat());
+};
 ````
 

--- a/src/rules/noUndefinedTypes.js
+++ b/src/rules/noUndefinedTypes.js
@@ -28,7 +28,15 @@ const globalTypes = [
   'globalThis', 'global', 'window', 'self',
 ];
 
+const iterableIterator = [
+  'Iterable',
+  'Iterator',
+  'IteratorObject',
+];
+
 const typescriptGlobals = [
+  ...iterableIterator,
+
   // https://www.typescriptlang.org/docs/handbook/utility-types.html
   'Awaited',
   'Partial',

--- a/test/rules/assertions/noUndefinedTypes.js
+++ b/test/rules/assertions/noUndefinedTypes.js
@@ -2193,5 +2193,17 @@ export default /** @type {import('../index.js').TestCases} */ ({
         parser: typescriptEslintParser,
       },
     },
+    {
+      code: `
+        /**
+         * @param {...Iterable<any>} its
+         * @returns {IteratorObject<any>}
+         * @see https://developer.mozilla.org/Web/JavaScript/Reference/Global_Objects/Iterator/concat
+         */
+        const concatIteratorJSDoc = (...its) => {
+            return Iterator.from(its.flat());
+        };
+      `,
+    },
   ],
 });

